registry-client.js

```diff
@@ -204,6 +223,50 @@ this.Registry = (function (Promise, XMLHttpRequest, DOMParser, undefined) {
     })
   }
 
+  /**
+   * Obtain the applications URL for the given URI.
+   *
+   * @param {String} uri   The URI to look up.
+   * @returns {Promise}
+   */
+  Registry.prototype.getApplicationURL = function (uri) {
+    const self = this
+    return new Promise(function (resolve, reject) {
+      self
+          .getApplicationsEndpoints()
+          .then(function (request) {
+            let applicationURL
+            const asciiOutput = request.responseText
+            const asciiLines = asciiOutput.split('\n')
+            for (let i = 0, all = asciiLines.length; i < all; i++) {
+              const nextLine = asciiLines[i]
+              if (Registry.LINE_CHECKER.test(nextLine)) {
+                const keyValue = nextLine.split('=')
+                const key = keyValue[0].trim()
+                if (key === uri) {
+                  applicationURL = keyValue[1].trim()
+                  break
+                }
+              }
+            }
+
+            if (!applicationURL) {
+              reject({
+                uri: uri,
+                error: new Error('No such URI ' + uri)
+              })
+            } else {
+              resolve(applicationURL)
+            }
+          })
+          .catch(function (err) {
+            console.error(
+                'Error obtaining application URL > ' + (err.error ? err.error : err)
+            )
+          })
+    })
+  }
```

Comment:
Much of this is duplicate code from `getCapabilityURL()`.  Please isolate the common logic.

Reply:
There's about 10 lines of code I can see to make common (parsing out the url from the ajax response.)   If I add a function to the prototype there seems to be name spacing issues trying to use it .then sections. Ideas on how best to handle that?